x-pack/solutions/security/packages/kbn-securitysolution-utils/src/time_duration/convert_time_duration_to_ms.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { parseTimeDuration } from './parse_time_duration';
+
+/**
+ * Tries to convert time duration `{value}{unit}` (e.g. 3m, 5h, 6s) to milliseconds.
+ * Supports
+ *  - `s` seconds, e.g. 3s, 0s, -5s
+ *  - `m` minutes, e.g. 10m, 0m
+ *  - `h` hours, e.g. 7h
+ *  - `d` days, e.g. 3d
+ *
+ * Returns `undefined` when unable to perform conversion.
+ */
+export function convertTimeDurationToMs(input: string): number | undefined {
+  const duration = parseTimeDuration(input);
```

Comment:
Shouldn't the input be `TimeDuration`? Please find an explanation in another comment.

```suggestion
export function convertTimeDurationToMs(input: TimeDuration): number {
```